### PR TITLE
Solved issue #749

### PR DESCRIPTION
**Pull request by ath34-tech**
I simply look into server.py gather_envs function and found out that this function is exactly what is needed to be added so I created a copy of that function in __init__.py and return a list from that function.
## Description
I added a function called get_all_env, which is similar to server.py's gather_envs functions, it simply return the name of all envs in list format. In get_all_env function I returned a sorted list which return name of all env.
## Motivation and Context
It solves issue #749 

## How Has This Been Tested?
I simply create a script that was reproducing issue(that script is told in issue I mentioned) :
`from visdom import Visdom

#  instantiate the window class 
```
vis = Visdom(offline=True, log_to_filename="./log.visdom")
print(vis.get_env_list())`

```
And output for this:
`
{'env list': "['main', 'main12', 'main13', 'main15', 'main2', 'main3', 'main6', 'myown']"}
`

### CODE REVIEW

File: py/visdom/__init__.py [hunk 1]
```diff
@@ -679,8 +692,9 @@ def _send(self, msg, endpoint='events', quiet=False, from_log=False, create=True
 
         if self.offline:
             # If offline, don't even try to post
-            return msg['win'] if 'win' in msg else True
-
+            list_of_env=self.get_all_env()
+            return msg['win'] if 'win' in msg else json.dumps({"env list":f'{list_of_env}'})
```
Comment (JackUrb) [inline py/visdom/__init__.py]: You shouldn't be making this change in the `_send` call, as our expected behavior in offline mode is that sends return `True`. Instead the change should be not calling `_send` to begin with from in the `get_env_list` method.

File: py/visdom/static/js/layout-bin-packer.js.map [hunk 2]
```diff
@@ -0,0 +1 @@
...5B,EAwDA,EAAE,IAAI,MAAM,KAAK,CAAC,IAAI,KAAK,GAAG,MAAM,EAAE;AAxDtC,EAyDA,IAAI,UAAU,CAAC,MAAM,EAAE,KAAK,CAAC;AAzD7B,EA0DA;;AA1DA,EA4DA,EAAE,IAAI,OAAO,GAAG,IAAI,CAAC,KAAK,CAAC,KAAK,GAAG,IAAI,CAAC,YAAY,CAAC,KAAK,CAAC,CAAC,IAAI,CAAC;;AA5DjE,EA8DA,EAAE,IAAI,CAAC,GAAG,KAAK,GAAG,OAAO,GAAG,IAAI,CAAC,YAAY,CAAC,KAAK,CAAC,GAAG,CAAC;AA9DxD,EA+DA,EAAE,IAAI,CAAC,GAAG,IAAI,CAAC,KAAK,CAAC,KAAK,GAAG,OAAO,CAAC,GAAG,IAAI,CAAC,aAAa,CAAC,KAAK,CAAC;;AA/DjE,EAiEA,EAAE,OAAO,CAAC,CAAC,CAAC,CAAC,EAAE,CAAC,CAAC,CAAC,CAAC;AAjEnB,EAkEA,CAAC;;AAlED,EAoEA,SAAS,CAAC,SAAS,CAAC,MAAM,GAAG,UAAU,YAAY,EAAE;AApErD,EAqEA,EAAE,IAAI,OAAO,YAAY,KAAK,QAAQ,EAAE;AArExC,EAsEA,IAAI,MAAM,IAAI,SAAS,CAAC,8DAA8D,CAAC;AAtEvF,EAuEA;AAvEA,EAwEA,EAAE,IAAI,MAAM,GAAG,IAAI,CAAC,MAAM,EAAE;AAxE5B,EAyEA,EAAE,IAAI,MAAM,KAAK,CAAC,EAAE,EAAE,OAAO,CAAC,CAAC;;AAzE/B,EA2EA,EAAE,IAAI,WAAW,GAAG,IAAI,CAAC,GAAG,CAAC,IAAI,CAAC,KAAK,CAAC,YAAY,CAAC,IAAI,CAAC,YAAY,CAAC,CAAC,CAAC,EAAE,CAAC,CAAC,CAAC;AA3E9E,EA4EA,EAAE,WAAW,GAAG,WAAW,GAAG,CAAC,GAAG,WAAW,GAAG,CAAC;AA5EjD,EA6EA,EAAE,IAAI,IAAI,GAAG,IAAI,CAAC,IAAI,CAAC,MAAM,CAAC,WAAW,CAAC;AA7E1C,EA8EA,EAAE,IAAI,WAAW,GAAG,IAAI,GAAG,IAAI,CAAC,aAAa,CAAC,CAAC,CAAC;;AA9EhD,EAgFA,EAAE,OAAO,WAAW;AAhFpB,EAiFA,CAAC,;;;;,;;"}
```
Comment (JackUrb) [inline py/visdom/static/js/layout-bin-packer.js.map]: You shouldn't be adding anything to the static directory. This and below files aren't relevant to this PR.

File: py/visdom/__init__.py [hunk 3]
```diff
@@ -503,6 +507,15 @@ def register_event_handler(self, handler, target):
...
         self.event_handlers[target] = []
 
+
+    def get_all_env(self):
+
+        if self.DEFAULT_ENV_PATH is not None:
+            items = [i.replace('.json', '') for i in os.listdir(self.DEFAULT_ENV_PATH) if '.json' in i]
+        else:
+            items = []
+        return items
```
Comment (JackUrb) [inline py/visdom/__init__.py]: This misses the mark - usually `visdom`'s client is run on a different machine than the `visdom` server, and as such the contents of `DEFAULT_ENV_PATH` will usually be empty on a client machine. You should instead be creating an `offline_env_list` or something like that in `__init__` if `offline`, and then in `send` if `offline`, check the `env` that visdom is trying to write to.

**Review by JackUrb — CHANGES_REQUESTED** [review]
Hi @ath34-tech, thanks for the attempt here. Unfortunately this implementation is missing the mark. I've left some comments inline to describe what we're really going for.

To find out more about what's being sent in `_send`, try adding some `print` statements and running the `demo.py` - you should be able to see where in the arguments you can pull `env` from to populate the local env list.

File: py/visdom/__init__.py (hunk 3 above)
Comment (ath34-tech) [reply py/visdom/__init__.py]: can you elaborate what you meant by check the "env" that visdom is trying to write to.

Comment (JackUrb) [reply py/visdom/__init__.py]: Check the args of what `visdom` is sending when calling `send` - an env can be included in every one of visdom's functions, like here: https://github.com/fossasia/visdom/blob/06f9cb22fdc4da2cb74f65faf0d6ec64d89e03fc/py/visdom/__init__.py#L884
It ends up being managed as an `eid`: https://github.com/fossasia/visdom/blob/06f9cb22fdc4da2cb74f65faf0d6ec64d89e03fc/py/visdom/__init__.py#L902
And send sets it to the default if none is provided:
https://github.com/fossasia/visdom/blob/master/py/visdom/__init__.py#L667-L668

**Review by ath34-tech — COMMENTED** [review]
Hey did you asked me to do something like this?

**Pull request closed without merging**